chain/chain/src/chain.rs

```diff
@@ -1206,6 +1277,63 @@ impl Chain {
         }
     }
 
+    /// Check if we can request chunks for this orphan. Conditions are
+    /// 1) Among the `NUM_ORPHAN_ANCESTORS_CHECK` immediate parents of the block at least one is
+    ///    accepted(is in store), call it `ancestor`
+    /// 2) The next block of `ancestor` has the same epoch_id as the orphan block
+    ///    (This is because when requesting chunks, we will use `ancestor` hash instead of the
+    ///     previous block hash of the orphan to decide epoch id
+    /// 3) All the immediate parents of the block after `ancestor` are either orphans or in
+    ///     `blocks_with_missing_chunks`
+    /// 4) Orphans that with outstanding missing chunks request has not exceed `MAX_ORPHAN_MISSING_CHUNKS`
+    /// 5) Orphans have missing chunks
+    pub fn should_request_chunks_for_orphan(
+        &mut self,
+        me: &Option<AccountId>,
+        orphan: &Block,
+    ) -> Option<OrphanMissingChunksInfo> {
+        if self.orphans_requested_missing_chunks.len() >= MAX_ORPHAN_MISSING_CHUNKS {
+            return None;
+        }
+        if self.orphans_requested_missing_chunks.contains(orphan.hash()) {
+            return None;
+        }
+        let mut block_hash = *orphan.header().prev_hash();
+        for _ in 1..NUM_ORPHAN_ANCESTORS_CHECK {
```

Comment:
is it intentional that this starts from 1? So we only loop `NUM_ORPHAN_ANCESTORS_CHECK - 1` times?

Reply:
Yeah it was intentional, but I was overthinking, iterating from 0 was correct